### DISCUSSION TITLE
[Question]: Live View Performance Delays (any Ideas Pls)

### DISCUSSION THREAD

**Discussion by ckglobalroaming** [blakeblackshear/frigate#13116] [Ask A Question]
### What is your question:

I went a bit nuts on my frigate machine at home.
I have AMD Rizen 5k with Nvidia Graphics Card and 2x Dual Coral TPU boards (4TPUs) running with 15 cameras. Its all NVME and the Container has 16GB ram assigned to it.

The live view display is soo good in 0.14.0 its awesome

### Metrics:
- AM4 R5k
- 16Gb SHM
- NVME
- 10Gb Fibre Network to Core and GB Cameras
- Ubuntu 24.04 and compose
- All of my cameras into Frigate are capped at 4FPS H264. 
- All of my TPUs are at idle with infrace of 5-7ms with 0.1-0.5% Usage
- CPU is at ~9-16%
- GPU is at ~5-8%
- All Detectors and Camera CPU are doing nothing.

Basically the machine is idling.

### Problem:
But the issue at hand is the the live view at most times is upto 45 seconds behind realtime. I can walk into the house and watch myself.

### Observation:
If i restart Friggate it gets better for a while then slows down a drifts out again.

Any guidance appreciated. This is not looking like a resource problem.

**Comment by hawkeye217** [comment]
When viewing your live streams and they are behind real time, approximately how long have they been running for?

**Reply by ckglobalroaming** [reply]
It seems progressive. at 6-12 hours it crawls out to +30 seconds.

**Reply by hawkeye217** [reply]
Makes sense when it's been streaming that long.

I have some improvements I'll be committing in a PR here in the next few days that a few other users have already reported that it has resolved this issue for them.

**Reply by hawkeye217** [reply]
If you'd like to test my latest build with the improvements, you can grab it here: `ghcr.io/hawkeye217/frigate:dev-eb2d6b5`

There will be a fair amount of logging in your browser console that won't be present in the official frigate release, but they help with debugging for now.
